### ISSUE TITLE
can't connect to SP-2101W

### ISSUE THREAD

**Issue by buttonger** [mwittig/pimatic-edimax#1]
While starting the piratic server the following message come up.
"Unable to get model info of device sp1: Error: No such device: check name, Retryin"
Any Idea what to do?

**Comment by mwittig** [comment]
From the README: "As part of the device definition you need to provide the `deviceName` and `password` which have been set using the EdiPlug app provided by Edimax. Note, the `deviceName` refers to the Name field of the plug settings." Please make sure you have the `deviceName` set to the name you have assigned using the EdiPlug app. By default, this should be "edimax".

**Comment by buttonger** [comment]
In plugins:

```
{
  "plugin": "edimax",
  "interval": 30
},
```

  "devices": [
    {
      "id": "sp1",
      "class": "EdimaxSmartPlug",
      "name": "extra",
      "deviceName": "extra",
      "host": "10.0.0.61",
      "password": "1234"
    }
  ],

I tried to connect manually thru terminal and it works.
Is there anything else?

Greets
Jens

**Comment by mwittig** [comment]
Which firmware version do you run?

**Comment by buttonger** [comment]
FW Version 1.04

Do you need additional information?

Am Dienstag, 29. Dezember 2015 schrieb Marcus Wittig :

**Comment by mwittig** [comment]
Thanks. The firmware version is the same I am using.

@buttonger Please update the plugin to v 0.3.1. I have extended the debug feature to enable protocol-level debugging. This should help to understand what's going on. Please enable debugging as follows and post the debug log (please use pastebin or alike):

```
{
      "plugin": "edimax",
      "interval": 30,
      "debug": true
    },
```

**Comment by buttonger** [comment]
Updated to 0.3.1 and started pimatic

ttp://pastebin.com/0fvGw46J

**Comment by mwittig** [comment]
Thanks.According to the log the request for the SmartPlugin system info was replied with an a empty reply. According to my findings, this only happens if the deviceName value does match the assigned to the Smartplug using the mobile app. Please double.check (extra space, lower vs uppercase letters?). 

Does switching the smartplug via pimatic works though?

**Comment by buttonger** [comment]
Thank you for your help. I had a closer look to the manually send xml. There i read the id edimax. I tried to change the device-settings in pimatic an I changed the name and the devicename to "edimax". Now it works an the error has gone.
Do you need more logs oder other information?

**Comment by mwittig** [comment]
@buttonger 

Ah, ok. Great, that it is working for you now.  Did you also change the name assigned to the SmartPlug with the mobile app? Does the mobile app still show "extra"?

**Comment by buttonger** [comment]
Yes, the mobile app is showing "extra".
I added two more of the smartplugs. And they have different names. But the application doesn't care about the names.

**Comment by mwittig** [comment]
Thank you very much for clarifying this and for your time. I did some research on this and I think I got this wrong. Apparently, the SMARTPLUG id attribute is "edimax" regardless of the name assigned to the plug. I'll provide a fix for this asap.

**Comment by mwittig** [comment]
Just reopened it. I'll update the ticket when the fix is available

**Comment by mwittig** [comment]
Release 0.3.2 should fix the issue

**Comment by mwittig** [comment]
@buttonger Did you try version 0.3.2? Does it work for you? Do you have Edimax SP 2101W which includes a power measurement and logging capabilities? If yes, does pimatic show the appropriate attribute values?

**Comment by mwittig** [comment]
@buttonger Fyi, I found another bug (see issue #1) which has been fixed with release 0.3.3

**Comment by buttonger** [comment]
I installied the update. I going to test later this week.

Von meinem Newton gesendet.
